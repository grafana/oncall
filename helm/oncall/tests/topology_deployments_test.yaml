suite: test topologySpreadConstraints for deployments
templates:
  - celery/deployment.yaml
  - engine/deployment.yaml
release:
  name: oncall
tests:
  - it: topologySpreadConstraints=[] -> should exclude topologySpreadConstraints
    asserts:
      - notExists:
          path: spec.template.spec.topologySpreadConstraints

  - it: topologySpreadConstraints -> should use custom topologySpreadConstraints
    set:
      engine:
        topologySpreadConstraints:
          - labelSelector:
              matchLabels:
                app.kubernetes.io/component: engine
            maxSkew: 1
            topologyKey: topology.kubernetes.io/zone
            whenUnsatisfiable: DoNotSchedule
      celery:
        topologySpreadConstraints:
          - labelSelector:
              matchLabels:
                app.kubernetes.io/component: engine
            maxSkew: 1
            topologyKey: topology.kubernetes.io/zone
            whenUnsatisfiable: DoNotSchedule
    asserts:
      - matchSnapshot:
          path: spec.template.spec.topologySpreadConstraints
