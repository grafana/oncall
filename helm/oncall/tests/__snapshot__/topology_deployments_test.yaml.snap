topologySpreadConstraints -> should use custom topologySpreadConstraints:
  1: |
    - labelSelector:
        matchLabels:
          app.kubernetes.io/component: engine
      maxSkew: 1
      topologyKey: topology.kubernetes.io/zone
      whenUnsatisfiable: DoNotSchedule
  2: |
    - labelSelector:
        matchLabels:
          app.kubernetes.io/component: engine
      maxSkew: 1
      topologyKey: topology.kubernetes.io/zone
      whenUnsatisfiable: DoNotSchedule
